### PR TITLE
fix(resolve): do not recursively apply alias mapping

### PR DESCRIPTION
**Pull request by kwonoj**
### Description

Attempt to fix WEB-1178, when there's an alias like ` "*" : [ "somepath/* "]` creates infinite recursion.

### CODE REVIEW

**Review by jridgewell — COMMENTED** [review]
Can we verify this against front or similar app?

**Comment by kwonoj** [comment]
I tried to use jsconfig test app which have some known patterns of alias. Any specific pattern would like to test more?

File: crates/turbopack-core/src/resolve/mod.rs [hunk 1]
```diff
@@ -731,8 +731,16 @@ async fn resolve_internal(
         let result_ref = import_map.lookup(context, request).await?;
         let result = &*result_ref;
         if !matches!(result, ImportMapResult::NoEntry) {
+            // Prevent alias recursively applied infinitely, i.e
+            // paths: { "*": ["node_modules/*"] }
+            let resolve_op = ResolveOptions {
+                import_map: None,
```
Comment (sokra) [inline crates/turbopack-core/src/resolve/mod.rs:737]: This should also prevent transitive aliases, which should be fine.

```
paths: {
  "a": ["b"],
  "b": ["c"]
}

Comment (kwonoj) [reply crates/turbopack-core/src/resolve/mod.rs:737]: ugh. any suggestions?

**Comment by kwonoj** [comment]
We'll need other approach probably.

**Pull request closed without merging**